Sources/Fuzzilli/Core/CodeGenerators.swift

```diff
@@ -527,18 +527,36 @@ public let CodeGenerators: [CodeGenerator] = [
     },
 
     CodeGenerator("SwitchCaseGenerator", input: .anything) { b, cond in
-        b.doSwitch(on: cond) { cases in
-            cases.addDefault {
-                b.generateRecursive()
-            }
-            for _ in 0..<Int.random(in: 0...5) {
-                cases.add(b.randVar(), fallsThrough: probability(0.1)) {
-                    b.generateRecursive()
+        var candidates: [Variable] = []
```

Comment:
Okay cool! Yeah I agree that this approach is better. I'd still like to be able to generate switch statements without a default case though, and there are a few other nits. How about we turn this function into

```swift
var candidates: [Variable] = []

// Generate a minimum of three cases (including a potential default case)
for _ in 0..<Int.random(in: 3...8) {
    candidates.append(b.randVar())
}

// If this is set, the selected candidate becomes the default case
var defaultCasePosition = -1
if probability(0.8) {
  defaultCasePosition = Int.random(in: 0..<candidates.count)
}

b.doSwitch(on: cond) { cases in
    for (idx, val) in candiates.enumerated() {
        if idx == defaultCasePosition {
            cases.addDefault(previousCaseFallsThrough: probability(0.1)) {
                b.generateRecursive()
            }
        } else {
            cases.add(val, previousCaseFallsThrough: probability(0.1)) {
                b.generateRecursive()
            }
        }
    }
}
```
I think that should then cover everything :D

Reply:
Nice one! I'll add these changes